plugins/Compressor/CompressorControlDialog.cpp

```diff
@@ -0,0 +1,757 @@
+		p.setCompositionMode(QPainter::CompositionMode_SourceOver);
+
+		p.end();
+
+		p.begin(&m_kneePixmap2);
+		
+		p.setCompositionMode(QPainter::CompositionMode_Source);
+		p.fillRect(0, 0, m_windowSizeX, m_kneeWindowSizeY, QColor("transparent"));
+		p.setCompositionMode(QPainter::CompositionMode_SourceOver);
+
+		p.end();
+
+		m_lastKneePoint = 0;
+	}
+
+	// Start drawing second knee layer
+	p.begin(&m_kneePixmap2);
+
+	p.setRenderHint(QPainter::Antialiasing, false);
+
+	int kneePoint = dbfsToXPoint(ampToDbfs(peakAvg));
+	if (kneePoint > m_lastKneePoint)
+	{
+		QRectF knee2Rect = QRect(m_lastKneePoint, 0, kneePoint - m_lastKneePoint, m_kneeWindowSizeY);
+		p.drawPixmap(knee2Rect, m_kneePixmap, knee2Rect);
+	}
+	else
+	{
+		p.setCompositionMode(QPainter::CompositionMode_Source);
+		p.fillRect(kneePoint, 0, m_lastKneePoint, m_kneeWindowSizeY, QColor("transparent"));
+		p.setCompositionMode(QPainter::CompositionMode_SourceOver);
+	}
+	m_lastKneePoint = kneePoint;
+
+	p.end();
+
+	if (m_controls->m_effect->m_redrawThreshold)
+	{
+		p.begin(&m_miscPixmap);
+
+		p.setCompositionMode(QPainter::CompositionMode_Source);
+		p.fillRect(0, 0, m_windowSizeX, m_windowSizeY, QColor("transparent"));
+		p.setCompositionMode(QPainter::CompositionMode_SourceOver);
+
+		p.setRenderHint(QPainter::Antialiasing, true);
+
+		// Draw threshold lines
+		p.setPen(QPen(m_threshColor, 2, Qt::DotLine));
+		p.drawLine(0, threshYPoint, m_windowSizeX, threshYPoint);
+		p.drawLine(threshXPoint, 0, threshXPoint, m_kneeWindowSizeY);
+
+		p.end();
+
+		m_controls->m_effect->m_redrawThreshold = false;
+	}
+
+	m_lastPoint = yPoint;
+	m_lastGainPoint = yGainPoint;
+
+	update();
+}
+
+
+
+void CompressorControlDialog::paintEvent(QPaintEvent *event)
+{
+	if (!isVisible())
+	{
+		return;
+	}
+
+	QPainter p(this);
+
+	p.setCompositionMode(QPainter::CompositionMode_Source);
+	p.fillRect(0, 0, m_windowSizeX, m_windowSizeY, QColor("transparent"));
+	p.setCompositionMode(QPainter::CompositionMode_SourceOver);
+
+	p.drawPixmap(0, 0, m_graphPixmap);
+	p.drawPixmap(0, 0, m_visPixmap);
+	p.setOpacity(0.25);
+	p.drawPixmap(0, 0, m_kneePixmap);
+	p.setOpacity(1);
+	if (m_controls->m_effect->isEnabled() && m_controls->m_effect->isRunning())
+	{
+		p.drawPixmap(0, 0, m_kneePixmap2);
+	}
+	p.drawPixmap(0, 0, m_miscPixmap);
+
+	p.end();
+}
+
+
+inline int CompressorControlDialog::dbfsToYPoint(float inDbfs)
+{
+	return (-((inDbfs + m_dbRange) / m_dbRange) + 1) * m_windowSizeY;
+}
+
+inline int CompressorControlDialog::dbfsToXPoint(float inDbfs)
+{
+	return m_kneeWindowSizeY - dbfsToYPoint(inDbfs);
+}
+
+
+void CompressorControlDialog::resizeEvent(QResizeEvent *event)
+{
+	resetCompressorView();
+}
+
+
+void CompressorControlDialog::wheelEvent(QWheelEvent * event)
+{
+	const float temp = m_dbRange;
+	m_dbRange = round(
+		qBound(COMP_GRID_SPACING, m_dbRange - copysignf(COMP_GRID_SPACING, event->delta()), COMP_GRID_MAX)
+		/ COMP_GRID_SPACING) * COMP_GRID_SPACING;
```

Comment:
Maybe pull the `qBound` out to a temp variable in order to make this format better. As it sits this is sort of hard to read.